elasticsearch/src/main/java/org/hibernate/search/elasticsearch/util/impl/GsonHttpEntity.java

```diff
@@ -169,14 +169,27 @@ public Header getContentEncoding() {
 	public InputStream getContent() throws IOException {
 		//This could be implemented but would be sub-optimal compared to using produceContent().
 		//We therefore prefer throwing the exception so that we can easily spot unintended usage via tests.
-		throw new UnsupportedOperationException( "Not implemented! Expected to produce content only over produceContent()" );
+		throw new UnsupportedOperationException( "Not implemented! Expected to produce content only over produceContent(),"
+				+ " or writeTo(OutputStream) if blocking calls are acceptable for your use case." );
 	}
 
 	@Override
-	public void writeTo(OutputStream outstream) throws IOException {
-		//This could be implemented but would be sub-optimal compared to using produceContent().
-		//We therefore prefer throwing the exception so that we can easily spot unintended usage via tests.
-		throw new UnsupportedOperationException( "Not implemented! Expected to produce content only over produceContent()" );
+	public void writeTo(OutputStream out) throws IOException {
+		/*
+		 * For this method we use no pagination, so ignore the mutable fields.
+		 *
+		 * Note we don't close the counting stream or the writer,
+		 * because we must not close the output stream that was passed as a parameter.
+		 */
+		CountingOutputStream countingStream = new CountingOutputStream( out );
+		Writer writer = new OutputStreamWriter( countingStream, CHARSET );
+		for ( JsonObject bodyPart : bodyParts ) {
+			gson.toJson( bodyPart, writer );
+			writer.append( '\n' );
+		}
+		writer.flush();
+		//Now we finally know the content size in bytes:
+		hintContentLenght( countingStream.getBytesWritten() );
```

Comment:
The code uses the number of bytes as a content length even before this patch.
As for the encoding, I'll have a look.